components/img_generator/img_generator.py

```diff
@@ -0,0 +1,167 @@
+import io
+from datetime import datetime
+from babel.dates import format_datetime
+from PIL import Image, ImageDraw, ImageSequence, ImageFont
+from protos.activity_id_pb2 import ActivityID
+from protos.rated_player_pb2 import RatedPlayer
+
+# Coordinates relative to (X=0, Y=0) for any section on the GIF.
+# Name is short to keep the logic compact in methods below.
+COORDS = {
+    "activity": (20, 20),
+    "details": (20, 40),
+    "players": [
+        (30, 60),
+        (30, 85),
+        (30, 110),
+        (30, 135),
+        (30, 160),
+        (30, 185)
+    ],
+    "substitutes": [
+        (220, 148),
+        (220, 168),
+        (220, 188)
+    ],
+    "section_w": 350,
+    "delta_x": 486,
+    "delta_y": 260,
+}
+
+ACTIVITY_NAMES = {
+    ActivityID.Type.LEVIATHAN: "Léviathan",
+    ActivityID.Type.EATER_OF_WORLDS: "Dévoreur de Mondes",
+    ActivityID.Type.SPIRE_OF_STARS: "Flèche d'étoiles",
+    ActivityID.Type.CROWN_OF_SORROW: "Couronne du Malheur",
+    ActivityID.Type.LAST_WISH: "Dernier Voeu",
+    ActivityID.Type.SCOURGE_OF_THE_PAST: "Fléau du Passé",
+    ActivityID.Type.GARDEN_OF_SALVATION: "Jardin du Salut",
+    ActivityID.Type.VAULT_OF_GLASS: "Caveau de Verre",
+    ActivityID.Type.CROPTAS_END: "Cropta",
+    ActivityID.Type.THE_TAKEN_KING: "La Chute du Roi",
+    ActivityID.Type.WRATH_OF_THE_MACHINE: "Fureur Mécanique",
+}
+
+# Add Prestige names when needed
+for activity_id in list(ACTIVITY_NAMES):
```

Comment:
Seems a bit more brittle to rely on the _PRESTIGE suffix but not too fussed about it.

Reply:
Removing it, it doesn't save that many lines anyway